test/tracer/propagation/cloud_trace_propagation_test.cpp

```diff
@@ -0,0 +1,110 @@
+#include <algorithm>
+
+#include "test/recorder/in_memory_recorder.h"
+#include "test/tracer/propagation/http_headers_carrier.h"
+#include "test/tracer/propagation/text_map_carrier.h"
+#include "test/tracer/propagation/utility.h"
+#include "tracer/legacy/legacy_tracer_impl.h"
+#include "tracer/tracer_impl.h"
+
+#include "3rd_party/catch2/catch.hpp"
+using namespace lightstep;
+
+TEST_CASE("cloud_trace propagation") {
+  LightStepTracerOptions tracer_options;
+  tracer_options.propagation_modes = {PropagationMode::cloud_trace};
+  auto recorder = new InMemoryRecorder{};
+  auto tracer = std::shared_ptr<opentracing::Tracer>{
+      new TracerImpl{MakePropagationOptions(tracer_options),
+                     std::unique_ptr<Recorder>{recorder}}};
+  std::unordered_map<std::string, std::string> text_map;
+  TextMapCarrier text_map_carrier{text_map};
+  HTTPHeadersCarrier http_headers_carrier{text_map};
+
+  SECTION("Inject, extract yields the same span context.") {
+    for (auto use_128bit_trace_ids : {true, false}) {
+      auto test_span_contexts = MakeTestSpanContexts(use_128bit_trace_ids);
```

Comment:
Can you add a case to MakeTestSpanContexts with traceids/spanid/spanid set to their maximum values `std::numeric_limits<uint64_t>::max()`?

Reply:
https://github.com/lightstep/lightstep-tracer-cpp/pull/248/commits/c9bd7abbed427e5d1f68833eb44926631fabe4bc

Thanks, that actually flagged up an issue that I'd estimated the potential maximum length of the header incorrectly (I didn't account for the null terminator)